src/kili/services/copy_project/__init__.py

```diff
@@ -119,3 +157,154 @@ def _copy_quality_settings(self, new_project_id: str, src_project: Dict) -> None
             use_honeypot=src_project["useHoneyPot"],
             review_coverage=src_project["reviewCoverage"],
         )
+
+    def _copy_assets(self, from_project_id: str, new_project_id: str, batch_size: int = 200):
+        """
+        Copy assets from a project to another.
+
+        Fetches assets by batch since `content` urls expire.
+        """
+        nb_assets_to_copy = self.count_assets(project_id=from_project_id)
+        skip = 0
+        with tqdm(total=nb_assets_to_copy) as pbar:
+            while skip < nb_assets_to_copy:
+                assets = self.assets(
+                    project_id=from_project_id,
+                    fields=["content", "externalId"],
+                    skip=skip,
+                    first=batch_size,
+                    disable_tqdm=True,
+                )
+                self.append_many_to_dataset(
+                    project_id=new_project_id,
+                    content_array=[x["content"] for x in assets],
+                    external_id_array=[x["externalId"] for x in assets],
+                    disable_tqdm=True,
+                )
+                skip += len(assets)
+                pbar.update(skip)
+
+    # pylint: disable=too-many-locals
+    def _copy_labels(self, from_project_id: str, new_project_id: str):
+        assets_new_project = self.assets(
+            new_project_id, fields=["externalId", "id"], disable_tqdm=True
+        )
+        assets_new_project = {asset["externalId"]: asset["id"] for asset in assets_new_project}
+
+        members_new_project = self.project_users(
+            project_id=new_project_id, fields=["user.email", "user.id"], disable_tqdm=True
+        )
+        members_new_project = {
+            member["user"]["email"]: member["user"]["id"] for member in members_new_project
+        }
+
+        nb_labels_to_copy = self.count_labels(project_id=from_project_id)
+        if nb_labels_to_copy == 0:
+            return
+
+        labels = self.labels(
+            project_id=from_project_id,
+            fields=[
+                "author.email",
+                "jsonResponse",
+                "secondsToLabel",
+                "isLatestLabelForUser",
+                "labelOf.externalId",
+                "labelType",
+                "modelName",
+            ],
+            disable_tqdm=True,
+        )
+        labels = [x for x in labels if x["isLatestLabelForUser"]]
```

Comment:
rename `x` by `label`

Reply:
Is there a reason why we only copy the last label of each user ?

Reply:
We cannot pass it to `append_labels` nor to `update_properties_in_label`